test/e2e/parallel/spacebindingrequest_test.go

```diff
@@ -0,0 +1,267 @@
+		spaceBindingRequest, err = memberAwait.WaitForSpaceBindingRequest(t, types.NamespacedName{Namespace: spaceBindingRequest.GetNamespace(), Name: spaceBindingRequest.GetName()},
+			UntilSpaceBindingRequestHasConditions(spacebindingrequesttestcommon.Ready()),
+			UntilSpaceBindingRequestHasSpecSpaceRole(spaceBindingRequest.Spec.SpaceRole),
+			UntilSpaceBindingRequestHasSpecMUR(newmur.GetName()), // new MUR
+		)
+		require.NoError(t, err)
+		_, err = hostAwait.WaitForSpaceBinding(t, spaceBindingRequest.Spec.MasterUserRecord, space.Name,
+			UntilSpaceBindingHasMurName(newmur.GetName()), // has new MUR
+			UntilSpaceBindingHasSpaceName(space.Name),
+			UntilSpaceBindingHasSpaceRole(spaceBindingRequest.Spec.SpaceRole),
+		)
+		require.NoError(t, err)
+	})
+}
+
+func NewSpaceBindingRequest(t *testing.T, awaitilities Awaitilities, memberAwait *MemberAwaitility, hostAwait *HostAwaitility, spaceRole string) (*toolchainv1alpha1.Space, *toolchainv1alpha1.SpaceBindingRequest, *toolchainv1alpha1.SpaceBinding) {
+	space, _, _ := CreateSpace(t, awaitilities, testspace.WithTierName("appstudio"), testspace.WithSpecTargetCluster(memberAwait.ClusterName))
+	// wait for the namespace to be provisioned since we will be creating the SpaceBindingRequest into it.
+	space, err := hostAwait.WaitForSpace(t, space.Name, UntilSpaceHasAnyProvisionedNamespaces())
+	require.NoError(t, err)
+	// let's create a new MUR that will have access to the space
+	username := uuid.Must(uuid.NewV4()).String()
+	_, mur := NewSignupRequest(awaitilities).
+		Username(username).
+		Email(username + "@acme.com").
+		ManuallyApprove().
+		TargetCluster(memberAwait).
+		RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+		NoSpace().
+		WaitForMUR().Execute(t).Resources()
+	// create the spacebinding request
+	spaceBindingRequest := CreateSpaceBindingRequest(t, awaitilities, memberAwait.ClusterName,
+		WithSpecSpaceRole(spaceRole),
+		WithSpecMasterUserRecord(mur.GetName()),
+		WithNamespace(GetDefaultNamespace(space.Status.ProvisionedNamespaces)),
+	)
+
+	// then
+	// check for the spaceBinding creation
+	spaceBinding, err := hostAwait.WaitForSpaceBinding(t, spaceBindingRequest.Spec.MasterUserRecord, space.Name,
+		UntilSpaceBindingHasMurName(spaceBindingRequest.Spec.MasterUserRecord),
+		UntilSpaceBindingHasSpaceName(space.Name),
+		UntilSpaceBindingHasSpaceRole(spaceBindingRequest.Spec.SpaceRole),
+		UntilSpaceBindingHasLabel(toolchainv1alpha1.SpaceBindingRequestLabelKey, spaceBindingRequest.GetName()),
+		UntilSpaceBindingHasLabel(toolchainv1alpha1.SpaceBindingRequestNamespaceLabelKey, spaceBindingRequest.GetNamespace()),
+	)
+	require.NoError(t, err)
+	// wait for spacebinding request status
+	spaceBindingRequest, err = memberAwait.WaitForSpaceBindingRequest(t, types.NamespacedName{Namespace: spaceBindingRequest.GetNamespace(), Name: spaceBindingRequest.GetName()},
+		UntilSpaceBindingRequestHasConditions(spacebindingrequesttestcommon.Ready()),
+	)
+	require.NoError(t, err)
```

Comment:
but we could maybe check that everything was propagated to the NSTemplateSet correctly  here, WDYT?
something like this:
```go
	if spaceRole == "admin" {
		_, err = memberAwait.WaitForNSTmplSet(t, space.Name,
			UntilNSTemplateSetHasSpaceRoles(
				SpaceRole(tier.Spec.SpaceRoles[spaceRole].TemplateRef, userSignup.Status.CompliantUsername, userSignup.mur.Name)))
	} else {
		_, err = memberAwait.WaitForNSTmplSet(t, space.Name,
			UntilNSTemplateSetHasSpaceRoles(
				SpaceRole(tier.Spec.SpaceRoles["admin"].TemplateRef, userSignup.Status.CompliantUsername),
				SpaceRole(tier.Spec.SpaceRoles[spaceRole].TemplateRef, userSignup.mur.Name)))
	}
	VerifyResourcesProvisionedForSpace(t, awaitilities, space.Name)
```
you know - to be sure that the SpaceBinding that we created will be properly processed by the Space controller

Reply:
Ok I see what you meant , but I'm not 100% sure I understood your example code - why do we need to do it this way? 

If I'm not mistaken the usernames added to the NSTemplateSet are the just the `SpaceBinding.Spec.MasterUserRecord` : https://github.com/codeready-toolchain/host-operator/blob/723b3c9794f442315117b4fd55dc4eb028930246/controllers/space/space_controller.go#L458-L471

So in theory this could be just:
```
tier, err := awaitilities.Host().WaitForNSTemplateTier(t, space.Spec.TierName)
_, err = memberAwait.WaitForNSTmplSet(t, space.Name,
      wait.UntilNSTemplateSetHasSpaceRoles(
	wait.SpaceRole(tier.Spec.SpaceRoles[spaceRole].TemplateRef, mur.Name)))
require.NoError(t, err)
VerifyResourcesProvisionedForSpace(t, awaitilities, space.Name)
```

but I'm sure I'm missing something.

Reply:
I now understand why we need todo it this way, it's because we have 2 murs 🤦 . 
Done in https://github.com/codeready-toolchain/toolchain-e2e/pull/779/commits/88eaeb5080ea1d9a058b54dda7d45e98c1102305